src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -881,6 +889,19 @@ private static Conversion ClassifyNullLiteralConversion(BoundExpression source,
             return Conversion.NoConversion;
         }
 
+        private Conversion ClassifyDefaultConversion(BoundDefaultPendingInference source, TypeSymbol destination, ref HashSet<DiagnosticInfo> useSiteDiagnostics)
+        {
+            Debug.Assert((object)source != null);
+            Debug.Assert((object)destination != null);
+
+            if (destination.IsStructType() && !destination.IsNullableType())
+            {
```

Comment:
In (at least one of) the corresponding language proposal, there is no such restriction. Not sure if you are proposing to add this as a restriction? Probably requires LDM discussion.